static/app/views/insights/common/views/spans/selectors/domainSelector.tsx

```diff
@@ -87,24 +89,39 @@ export function DomainSelector({
   });
 
   const incomingDomains = [
-    ...uniq(domainData?.flatMap(row => row[SpanMetricsField.SPAN_DOMAIN])),
+    ...uniq(
+      domainData?.flatMap(row => {
+        const spanDomain: string | string[] = row[SpanMetricsField.SPAN_DOMAIN];
+        // `useInsightsEap` returns a string, but `metrics` returns an array
+        if (typeof spanDomain === 'string') {
+          return spanDomain.split(',');
+        }
+        return spanDomain;
+      })
+    ).filter(Boolean),
   ];
 
+  const domainList = incomingDomains.map(domain => ({
+    label: domain,
+    value: useEap ? `*${domain}*` : domain,
+  }));
+
   if (value) {
-    incomingDomains.push(value);
+    const scrubbedValue = useEap
+      ? value.replace(/^\*(.*)\*$/, '$1') // removes wildcards, only if they are at the beginning and end of the string
```

Comment:
Sorry, I'm not following, why is this needed?

Reply:
In the domain selector we add the current selected domain to the domain list,
`value` comes from the query params ` decodeScalar(location.query?.['span.domain'])`
i.e 
```
if (value {
domainList.push({
      label: scrubbedValue,
      value,
    });
}
```

In this case, when you select a domain, for example `table1`, the query param becomes `*table1*`.

If we added that back to the `domainList` directly, we would have an entirely new entry for `*table1*`. Therefore we need to remove the wildcards to prevent adding the wildcard version to the list.